modules/renter/hostdb/hostdb_test.go

```diff
@@ -3,58 +3,115 @@ package hostdb
 import (
 	"io/ioutil"
 	"os"
+	"path/filepath"
 	"testing"
 
 	"github.com/NebulousLabs/Sia/build"
 	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/modules/consensus"
+	"github.com/NebulousLabs/Sia/modules/gateway"
 	"github.com/NebulousLabs/Sia/modules/renter/hostdb/hosttree"
 	"github.com/NebulousLabs/Sia/persist"
+	"github.com/NebulousLabs/Sia/types"
 )
 
+type hdbTester struct {
+	cs      modules.ConsensusSet
+	gateway modules.Gateway
+
+	hdb *HostDB
+
+	persistDir string
+}
+
 // bareHostDB returns a HostDB with its fields initialized, but without any
 // dependencies or scanning threads. It is only intended for use in unit tests.
 func bareHostDB() *HostDB {
 	hdb := &HostDB{
 		log: persist.NewLogger(ioutil.Discard),
 
-		activeHosts: make(map[modules.NetAddress]*hostEntry),
-		allHosts:    make(map[modules.NetAddress]*hostEntry),
-		scanPool:    make(chan *hostEntry, scanPoolSize),
+		scanPool: make(chan modules.HostDBEntry, scanPoolSize),
 	}
 	hdb.hostTree = hosttree.New(hdb.calculateHostWeight)
 	return hdb
 }
 
 // newStub is used to test the New function. It implements all of the hostdb's
 // dependencies.
-type newStub struct{}
+type newStub struct {
+	*gateway.Gateway
```

Comment:
I am in the process of pulling all of the stubs out of the hostdb, they make it very high-friction to change any internals.